.github/workflows/linux.yaml

```diff
@@ -97,21 +97,36 @@ jobs:
       - name: Download Source Package
         uses: actions/download-artifact@v3
         with:
-            name: Sources
+          name: Sources
 
       - name: Install Build Dependencies
         run: |
+          # Install and build OpenSSL from scratch.
+          #
+          # Using `yum install openssl` doesn't seem to make cmake happy,
+          # so we have to do it the hard way.
+          yum -y install perl git
+          git clone https://github.com/openssl/openssl.git
+          cd openssl
+          ./Configure
+          make
+          # I think the issue with `yum install` is that it will install the package
+          # in a folder Cmake is not expecting, while this command always users /usr/local
+          make install
+          cd ..
```

Comment:
I tried `dnf` with `--installroot` because `yum` doesn't have an option like that (that I could find). However didn't have success with that -- also didn't try very hard, gotta be honest.

Anyways, the main issue is that this makes this specific CI job exceedingly slow. Maybe we can cache the OpenSSL build skip the build which is the longest running script? Let me know.

Reply:
Talked offline, que tl;dr; is: no this is not a good workaround, but we can fix it in a follow-up. Filed: https://mozilla-hub.atlassian.net/browse/VPN-3116?atlOrigin=eyJpIjoiY2NkMDQyZTVjNmU0NDYzZjk4ZWI5NmY0N2RmYzc4NzAiLCJwIjoiamlyYS1zbGFjay1pbnQifQ